doc/user/content/transform-data/optimization.md

```diff
@@ -49,20 +49,20 @@ EXPLAIN SELECT * FROM foo WHERE x = 42 AND y = 'hello';
      ReadExistingIndex materialize.public.foo_x_y lookup_value=(42, "hello")+
                                                                             +
  Used Indexes:                                                              +
-   - materialize.public.foo_x_y                                             +
+   - materialize.public.foo_x_y (lookup)                                    +
 ```
 
 #### Matching multi-column indexes to multi-column `WHERE` clauses
 
 In general, your index key should exactly match the columns that are constrained in the `WHERE` clause. In more detail:
-- If the `WHERE` clause constrains fewer fields than your index key includes, then **the index will not be used**. For example, an index on `(x, y)` cannot be used to speed up `WHERE x = 7`.
+- **_If the `WHERE` clause constrains fewer fields than your index key includes, then the index will not be used_** for a lookup, but will be fully scanned. For example, an index on `(x, y)` cannot be used to execute `WHERE x = 7` as a point lookup.
 - If the `WHERE` clause constrains more fields than your index key includes, then the index might still provide some speedup, but it won't necessarily be optimal: In this case, the index lookup is performed using only those constraints that are included in the index key, and the rest of the constraints will be used to subsequently filter the result of the index lookup.
 - If `OR` is used and not all arguments constrain the same fields, create an index for the intersection of the constrained fields. For example, if you have `WHERE (x = 51 AND y = 'bbb') OR (x = 76 AND z = 9)`, create an index just on `x`.
 - If `OR` is used and its arguments constrain completely disjoint sets of fields (e.g. `WHERE x = 5 OR y = 'aaa'`), try to rewrite your query using a `UNION` (or `UNION ALL`), where each argument of the `UNION` has one of the original `OR` arguments.
 
 ### `JOIN`
 
-In general, you can improve the performance of your joins by creating indexes on the columns being joined. This comes at the cost of additional memory usage. Fortunately, Materialize's in-memory [arrangements](/overview/arrangements) allow the system to share indexes across queries, which means **for multiple queries, an index is a fixed upfront cost with memory savings for each new query that uses it.**
+In general, you can improve the performance of your joins by creating indexes on the columns occurring in join keys. This comes at the cost of additional memory usage. Fortunately, Materialize's in-memory [arrangements](/overview/arrangements) (the internal data structure of indexes) allow the system to share indexes across queries, which means **for multiple queries, an index is a fixed upfront cost with memory savings for each new query that uses it.**
```

Comment:
nit: you can reference the joins blog post here.

Reply:
Other documents use the

```text
{{< warning >}}
Some highlighted text
{{< warning >}}
```

convention to highlight important information, but let's see what @morsapaes has to say about this.

Reply:
Done